lib/AST/FrontendSourceFileDepGraphFactory.cpp

```diff
@@ -188,22 +184,22 @@ std::string
 DependencyKey::computeNameForProvidedEntity<NodeKind::topLevel,
                                             PrecedenceGroupDecl const *>(
     const PrecedenceGroupDecl *D) {
-  return ::getName(D);
+  return D->getName().str().str();
 }
 template <>
 std::string DependencyKey::computeNameForProvidedEntity<
     NodeKind::topLevel, FuncDecl const *>(const FuncDecl *D) {
-  return ::getName(D);
+  return getBaseName(D);
 }
 template <>
 std::string DependencyKey::computeNameForProvidedEntity<
     NodeKind::topLevel, OperatorDecl const *>(const OperatorDecl *D) {
-  return ::getName(D);
+  return D->getName().str().str();
 }
 template <>
 std::string DependencyKey::computeNameForProvidedEntity<
     NodeKind::topLevel, NominalTypeDecl const *>(const NominalTypeDecl *D) {
-  return ::getName(D);
+  return D->getName().str().str();
```

Comment:
I don't think this part of the change is correct. The Identifier portion of special decl names is going to be NULL (or whatever interned markers we choose to use here in the future), which means these keys are going to wind up empty. You're going to want to recover the old behavior by using the `userFacingName()` of the base of the full name.

Reply:
Unless I'm misunderstanding the type of the member access here...

Reply:
Nominal type decl names are always Identifiers, so `userFacingName()` is equivalent to `getIdentifier().str()`.

Reply:
Makes sense.

Reply:
It may be more robust to use `userFacingName()` in case we ever need special names for nominals, but if that happens, `getName()`, which currently returns an Identifier for a nominal, would have to change anyway.